benchmarks/BM_efficientnet/README.md

```diff
@@ -0,0 +1,91 @@
+# Classification end-to-end benchmark
+
+## Overview
+This is a benchmark that presents end-to-end classification use-case. Composing parts of this
+benchmark are:
+1. GPU-powered data preprocessing using DALI,
+2. Image classification using EfficientNet,
+3. Inference deployment for Triton Inference Server.
+
+## How to run the benchmark?
+Running the benchmark requires three steps:
+1. Setup
+2. Run
+3. (optional) Tuning
+
+In the `Setup` step, the script downloads the EfficientNet model, converts it to TensorRT format
+and puts together a model repository and all the configuration files necessary for
+Triton Inference Server. It needs to be run only once for a given rig.
+
+In the `Run` step, the script runs a Triton's `perf_analyzer` to benchmark the use-case.
+There is a handful of parameters that can be tuned for the benchmark, you'd most probably
+like to turn these nobs in the `Tuning` step.
+
+Lastly, the `Tuning` step involves polishing the use-case parameters to find the optimal setup
+for a given hardware configuration. Details about tuning the solution are presented
+in the relevant section below.
+
+### Setup + run
+Running both `Setup` and `Run` is required the first time the benchmark is conducted on a given rig.
+Please use the following command and provide the maximal expected batch size (please refer to
+the section below for more information on the max batch size):
+```bash
+$ cd <dali_backend-repo-path>/benchmarks/BM_efficientnet
+$ bash benchmark.sh do_setup MAX_BATCH_SIZE
+
+e.g.
+$ cd /home/myuser/Triton/dali_backend/benchmarks/BM_efficientnet
+$ bash benchmark.sh do_setup 32
+```
+
+### Run
+Running the benchmark without the `Setup` step is possible with the command below. Please note
+that the script will assume that the benchmark is properly initialized. If not, unpredictable
+things may happen. You've been warned.
+```bash
+$ cd <dali_backend-repo-path>/benchmarks/BM_efficientnet
+$ bash benchmark.sh
+```
+
+### Factory reset
+Factory resetting benchmark might come in handy if an error occurred during the `Setup` step.
+To reset the benchmark, please run:
+```bash
+$ cd <dali_backend-repo-path>/benchmarks/BM_efficientnet
+$ bash reset_benchmark.sh 
+```
+
+## Tuning the inference
+Since this benchmark is targeted into DALI's usage in inference, we'll be focusing on tuning DALI
+rather than the EfficientNet model. For the information about the latter, please consult [Model Navigator documentation](https://triton-inference-server.github.io/model_navigator/0.7.1/).
+
+To obtain optimal performance numbers for your given hardware configuration, the scenario has
+to be tuned. DALI provides handful of parameters to manipulate. Here we are presenting details
+about where these parameters can be found. More information about the parameters themselves
+may be found in the [DALI documentation](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/index.html).
+
+You may tune the benchmark in the following places:
```

Comment:
I read only the readme file, but would it make sense to have those parameter adjustable from some ./run_benchmark.py/sh file instead of having to dig through all the configs. Or we want to teach the user into looking into those spots?

Reply:
It would not make much sense, since some of these parameters are set up in setup ;) For the sake of making these scripts easier, I'm assuming user will know how to look there. If he's about to deploy his solution to Triton, he would need to learn it anyway.